### PR TITLE
Adds defaultPaymentInstrumentId and method to customer (GWY-4262)

### PR DESCRIPTION
**Pull request by maxyengine**
Resolves: GWY-4262

### CODE REVIEW

File: spec/definitions/Customer.yaml [hunk 1]
```diff
@@ -24,6 +24,13 @@ properties:
     description: The customer default card ID
     allOf:
       - $ref: "#/definitions/ResourceId"
+  defaultPaymentInstrumentId:
+    description: The customer default payment instrument ID
+    allOf:
+      - $ref: "#/definitions/ResourceId"
+  defaultPaymentMethod:
+    description: The customer default payment method
+    $ref: "#/definitions/Method"
```
Comment (slavcodev) [inline spec/definitions/Customer.yaml]: Wrong spec here, by spec all fields except `$ref` is ignoring. You can see the right format in `defaultPaymentInstrumentId`

**Comment by slavcodev** [comment]
Adding `defaultPaymentMethod` and `defaultPaymentInstrumentId` means deprecated of `defaultCardId`, I think we need remove it form specs.

File: spec/definitions/Customer.yaml [hunk 2]
```diff
@@ -20,10 +20,13 @@ properties:
     description: The customer IP
     type: string
     format: ipv4
-  defaultCardId:
-    description: The customer default card ID
-    allOf:
-      - $ref: "#/definitions/ResourceId"
+  defaultPaymentInstrument:
+    description: The customer default payment instrument
```
Comment (adamaltman) [inline spec/definitions/Customer.yaml]: The `customer's` default payment instrument

File: spec/definitions/PaymentInstument/PayPalPaymentInstrument.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+type: object
+description: Ach payment instrument object
```
Comment (adamaltman) [inline spec/definitions/PaymentInstument/PayPalPaymentInstrument.yaml]: PayPal

File: spec/definitions/PaymentInstument/AchPaymentInstrument.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+type: object
+description: Ach payment instrument object
```
Comment (adamaltman) [inline spec/definitions/PaymentInstument/AchPaymentInstrument.yaml]: ACH

File: spec/definitions/Customer.yaml [hunk 5]
```diff
@@ -20,10 +20,13 @@ properties:
...
     type: string
     format: ipv4
-  defaultCardId:
-    description: The customer default card ID
-    allOf:
-      - $ref: "#/definitions/ResourceId"
+  defaultPaymentInstrument:
+    description: The customer's default payment instrument
+    type: object
+    enum:
```
Comment (slavcodev) [inline spec/definitions/Customer.yaml]: Hmm, interesting variant w/o discriminator 👍

File: spec/definitions/PaymentInstument/PaymentCardPaymentInstrument.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+description: Payment card payment instrument object
+required:
+  - method
+  - paymentCardId
+properties:
+  method:
+    description: The payment method
+    type: string
+    enum:
+      - payment_card
```
Comment (adamaltman) [inline spec/definitions/PaymentInstument/PaymentCardPaymentInstrument.yaml]: This was changed to `payment-card`, right?

Comment (maxyengine) [inline spec/definitions/PaymentInstument/PaymentCardPaymentInstrument.yaml]: yes but then checkout tests fail

**Comment by adamaltman** [comment]
Perhaps we should add the `CashInstrument` too

```
{
   "method": "cash"
}
```
(no other properties)

**Pull request closed without merging**